src/gnn_tracking/preprocessing/point_cloud_builder.py

```diff
@@ -57,9 +56,9 @@ def __init__(
         """
         # create outdir if necessary
         os.makedirs(outdir, exist_ok=True)
-        self.outdir = outdir
```

Comment:
I'd say this should be `Path(..)` (there's no reason not to have writing support here)

Reply:
I noticed in the function declaration that `outdir: str | PurePath`. `PurePath` seems to be a more specific type of `PathLike` objects, so I used the following:

```py
self.outdir = PurePath(outdir)
```

Reply:
no, `Path` is a subclass of `PurePath` (see [docs](https://docs.python.org/3/library/pathlib.html)), so it is more specific than the latter. We want to accept a broad range of inputs (actually `outdir: str| os.PathLike` would've been even better ) and then convert it to a `Path`

Reply:
Sure. I'll update.